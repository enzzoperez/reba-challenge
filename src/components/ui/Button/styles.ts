import {StyleSheet} from 'react-native';

interface StylesProps {
  textColor?: string;
  backgroundColor?: string;
}

export const styles = ({backgroundColor, textColor}: StylesProps) =>
  StyleSheet.create({
    rootButton: {
      justifyContent: 'center',
      alignItems: 'center',
      height: 32,
      width: 128,
      backgroundColor,
      borderRadius: 10,
    },
    textButton: {
      color: textColor,
    },
  });
