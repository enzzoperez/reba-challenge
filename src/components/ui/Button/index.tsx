import {useTheme} from '@react-navigation/native';
import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';

interface ButtonProps {
  label: string;
  onPress: () => {} | undefined;
}

const Button = ({label, onPress}: ButtonProps) => {
  const {colors} = useTheme();
  return (
    <TouchableOpacity
      onPress={() => onPress()}
      style={styles({backgroundColor: colors.primary}).rootButton}>
      <Text style={styles({textColor: colors.card}).textButton}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Button;
