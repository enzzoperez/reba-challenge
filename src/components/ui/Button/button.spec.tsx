import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './';

const defaultProps = {
  onPress: jest.fn(),
  label: 'Navigate',
};

it('renders correctly', () => {
  const tree = renderer.create(<Button {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
