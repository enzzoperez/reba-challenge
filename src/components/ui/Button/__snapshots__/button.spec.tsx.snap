// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(0, 122, 255)",
      "borderRadius": 10,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "width": 128,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
      }
    }
  >
    Navigate
  </Text>
</View>
`;
